### ISSUE TITLE
Issue on training with ready-made dataset

### ISSUE THREAD

**Issue by rose-jinyang** [XiaohangZhan/cdp#19]
Hello
Thanks for contributing this paper and code.
I had a successful inference with ready-made emore_u200k data and pre-trained CDP model.
I tried to retrain CDP model with ready-made emore_l200k in mediator mode.
But I met the following issue.
![image](https://user-images.githubusercontent.com/47862419/80086745-bbbf1800-858c-11ea-9a2e-4b2bf79ede64.png)
I am using a Tesla V100(GPU memory 16GB) and DDR 128GB.
PyTorch version is 1.4.0 and Python version is 3.7
I used the following config.yml.
![image](https://user-images.githubusercontent.com/47862419/80087389-97b00680-858d-11ea-962d-21fd0ab626f5.png)
I thought that this may be due to lack of GPU memory so I decreased the batch_size from 1024 to 8 but the result was same.
Please let me know the reason asap.
Thanks

**Comment by rose-jinyang** [comment]
Oh, I fixed the issue.
This was due to PyTorch version.
I changed Line 96 in mediator.py as the following
losses.update(loss.mean().data.cpu().item())
It started to train.

**Comment by XiaohangZhan** [comment]
> Oh, I fixed the issue.
> This was due to PyTorch version.
> I changed Line 96 in mediator.py as the following
> losses.update(loss.mean().data.cpu().item())
> It started to train.

Glad to hear that. The PyTorch version for me was 0.3.1.

**Comment by rose-jinyang** [comment]
Hi
I have a question.
I'd like to know about threshold in detail.
![image](https://user-images.githubusercontent.com/47862419/80166080-a0492100-860f-11ea-84d2-935372aba005.png)
Is this irrelative to face recognition model ?
Is this the Euclidean distance threshold between feature vectors ?
Please let me know asap.
Thanks

**Comment by XiaohangZhan** [comment]
Hi, the threshold is the binary classification threshold for MLP. It has nothing to do with the face recognition model. You may have to use a validation set to search this threshold.
